src/client/Files.js

```diff
@@ -501,21 +501,42 @@ exports.UploadFileData = async function({libraryId, objectId, writeToken, upload
   ValidateParameters({libraryId, objectId});
   ValidateWriteToken(writeToken);
 
-  let path = UrlJoin("q", writeToken, "file_jobs", uploadId, jobId);
+  let retries = 0;
+  do {
+    try {
+      const jobStatus = await this.UploadJobStatus({libraryId, objectId, writeToken, uploadId, jobId});
```

Comment:
Should we check here if "rem" is 0 - then we're done (in case we failed after finishing to upload, which can happen in congestion).